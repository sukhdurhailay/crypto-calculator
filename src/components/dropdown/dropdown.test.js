import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Dropdown from './dropdown.ui';

describe('Dropdown', () => {
  const props = {
    name: 'dropdown',
    options: [
      { value: 1, text: '1' },
      { value: 2, text: '2' },
      { value: 3, text: '3' }
    ],
    onChange: jest.fn()
  };

  it('renders without crashing', () => {
    shallow(<Dropdown {...props} />);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Dropdown {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
