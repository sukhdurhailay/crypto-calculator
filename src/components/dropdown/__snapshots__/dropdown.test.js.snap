// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown renders correctly 1`] = `
<div
  className="sc-bdVaJa LDTEX"
>
  <select
    className="sc-bwzfXH hJRDcZ"
    name="dropdown"
    onChange={[Function]}
  >
    <option
      value={1}
    >
      1
    </option>
    <option
      value={2}
    >
      2
    </option>
    <option
      value={3}
    >
      3
    </option>
  </select>
</div>
`;
